operator-pipeline-images/operatorcert/static_tests/common/catalog.py

```diff
@@ -0,0 +1,55 @@
+"""A common test suite for file based catalog"""
+
+from collections.abc import Iterator
+
+from operator_repo import OperatorCatalog, OperatorCatalogList
+from operator_repo.checks import CheckResult, Fail
+
+
+def _get_bundle_registries_from_catalog(catalog: OperatorCatalog) -> list[str]:
+    """Get bundle images from catalog"""
+    catalog_content = catalog.catalog_content
+    bundle_images = []
+    for item in catalog_content:
+        if isinstance(item, dict):
+            if item.get("schema") == "olm.bundle":
+                if image := item.get("image"):
+                    bundle_images.append(image)
+
+    return bundle_images
+
+
+def _filter_invalid_images(
+    bundle_images: list[str], allowed_registries: list[str]
+) -> list[str]:
+    """Filter out invalid images"""
+    invalid_images = []
+    for image in bundle_images:
+        if not any(image.startswith(registry) for registry in allowed_registries):
+            invalid_images.append(image)
+
+    return invalid_images
+
+
+def check_bundle_images_in_fbc(
```

Comment:
Should this file be renamed to `operator_catalogs.py` since it uses `OperatorCatalogList` as an input?

Reply:
This structure is kinda hard-coded to the [operator-repo](https://github.com/mporrato/operator-repo/blob/277471a08a7e9f06169925fcf25b039ac70c4677/src/operator_repo/checks/__init__.py#L84) library. I can change it there though so we can decide in the future if there will be any validation for the whole catalog.

Reply:
Fixed in external `operator-repo` and renamed.